### PR TITLE
Gold Price Prediction

### PR DESCRIPTION
**Pull request by Soumayan-pal01**
## Gold Price Prediction

- The aim of this project is to perform analysis the gold price using different EDA techniques, and eventually train different model to predict the price of gold.

- [x] LGM-SOC'21 Participant 
- [ ] Contributor

Closes: #258

### Describe the add-ons or changes you've made

In this project we have performed a detailed analysis and visualization of the training dataset with different Exploratory Data Analysis techniques. Then a comaprative analysis of different Regressors have been done to predict the gold price. After performing the model comparative analysis we can conclude that almost all the models had an accuracy above 99% except the ARIMA model, which did not perform well. The Random Forest Regressor and XGBoost regressor gave best results with 99.52% and 99.53% accuracy respectively.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by prathimacode-hub — APPROVED** [review]
Your PR is as valuable as gold. Extraordinary execution. 👌🎊🤩@Soumayan-pal01

**Comment by Soumayan-pal01** [comment]
> Your PR is as valuable as gold. Extraordinary execution. 👌🎊🤩@Soumayan-pal01

Thank you @prathimacode-hub .
